rxandroidble/src/main/java/com/polidea/rxandroidble/internal/operations/RxBleRadioOperationScan.java

```diff
@@ -46,8 +46,11 @@ public void run() {
     }
 
     public void stop() {
+
         // TODO: [PU] 29.01.2016 https://code.google.com/p/android/issues/detail?id=160503
-        rxBleAdapterWrapper.stopLeScan(leScanCallback);
+        if (startLeScanStatus) {
```

Comment:
It would ensure that status value is the same across different threads. Also it's more lightweight because we would avoid context switching.